habitat-hitl/habitat_hitl/core/remote_gui_input.py

```diff
@@ -11,12 +11,22 @@
 from habitat_hitl._internal.networking.average_rate_tracker import (
     AverageRateTracker,
 )
+from habitat_hitl._internal.networking.interprocess_record import (
+    InterprocessRecord,
+)
 from habitat_hitl.core.gui_input import GuiInput
+from habitat_hitl.core.key_mapping import KeyCode
 
 
 # todo: rename to RemoteClientState
 class RemoteGuiInput:
-    def __init__(self, interprocess_record, debug_line_render):
+    def __init__(
+        self,
+        interprocess_record: InterprocessRecord,
+        debug_line_render,
+        gui_input: GuiInput,
+    ):
```

Comment:
This is where the redirection happens. `RemoteGuiInput` used to create its own internal `GuiInput` - now it's passed as a dependency.